rust_snuba/src/strategies/clickhouse/batch.rs

```diff
@@ -48,7 +50,14 @@ impl BatchFactory {
             HeaderValue::from_str(database).unwrap(),
         );
 
-        let query_params = "load_balancing=in_order&insert_distributed_sync=1".to_string();
+        let mut query_params = String::new();
+        query_params.push_str("load_balancing=in_order&insert_distributed_sync=1");
+
+        let async_inserts_allowed = get_str_config("async_inserts_allowed").ok().flatten();
+        if async_inserts && async_inserts_allowed == Some(1.to_string()) {
```

Comment:
```suggestion
        if async_inserts && async_inserts_allowed == Some("1") {
```

that should also work

Reply:
I think I still need `to_string()` because it was expecting `String` instead of `&str`